File: 0136-single-number/my_approach.cpp
// Time complexity: O(1), Space complexity: O(1)
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int ans = nums[0];
        int len = nums.size();
        for (int i = 1; i < len; i++) {
            ans ^= nums[i];
        }
        return ans;
    }
};

File: 1-two-sum/my_approach.py
# Time complexity: O(N), Space complexity: O(N)
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        umap = {}
        size = len(nums)

        for i in range(size):
            num = nums[i]
            complement = target - num
            if complement in umap:
                return [umap[complement], i]
            umap[num] = i
        
        return []